tests/test_edgeql_expressions.py

```diff
@@ -834,6 +835,24 @@ class TestExpressions(tb.QueryTestCase):
             ],
         ])
 
+    async def test_edgeql_expr_array_15(self):
+        with self.assertRaisesRegex(
+                # FIXME: possibly a different error should be used here
+                exc.UnknownEdgeDBError,
+                r'array index 10 is out of bounds'):
+            await self.con.execute("""
+                SELECT [1, 2, 3][10];
+            """)
+
+    async def test_edgeql_expr_array_16(self):
+        with self.assertRaisesRegex(
+                # FIXME: possibly a different error should be used here
+                exc.UnknownEdgeDBError,
+                r'array index -7 is out of bounds'):
```

Comment:
Why does it say `index -7` when you're passing `-10`?

Reply:
That's an artefact of the fact that the conversion from negative index to "real" index happens before this step. So the original index is lost by this point in code. At this point it's possibly a good idea to have a generic negative index adjusting function to be used in `array_get` and `_array_index` that would allow preserving the original index value along side with the new adjusted value. As long as this isn't a terrible anti-pattern for SQL, I'd implement that.

Reply:
I've fixed it with a separate function that normalizes the indexes.